api/lib/infrastructure/repositories/knowledge-element-repository.js

```diff
@@ -88,18 +88,20 @@ module.exports = {
   },
 
   getSumOfPixFromUserKnowledgeElements(userId) {
-    return Bookshelf.knex.with('earnedPixWithRankPerSkill',
-      (qb) => {
-        qb.select('earnedPix', Bookshelf.knex.raw('ROW_NUMBER() OVER (PARTITION BY ?? ORDER BY ?? DESC) AS rank', ['skillId', 'createdAt']), 'competenceId')
-          .from('knowledge-elements')
-          .where({ userId });
-      })
-      .sum('earnedPix AS earnedPix')
-      .from('earnedPixWithRankPerSkill')
-      .where({ rank: 1 })
-      .groupBy('competenceId')
-      .then((pixEarnedByCompetence) => {
-        const pixScoreByCompetence = _.map(pixEarnedByCompetence, (pixEarnedForOneCompetence) =>  pixEarnedForOneCompetence.earnedPix);
+    const earnedPixByCompetenceQuery = `
+      WITH earnedPixWithRankPerSkill as (
+        select "earnedPix", "competenceId", ROW_NUMBER() OVER (PARTITION BY "skillId" order by "createdAt" DESC) AS rank
+        from "knowledge-elements"
+        where "userId"=??
+      )
+      SELECT "competenceId", sum("earnedPix") AS "earnedPix"
+      FROM earnedPixWithRankPerSkill
+      WHERE "rank" = 1
+      GROUP BY "competenceId"
+    `;
+    return Bookshelf.knex.raw(earnedPixByCompetenceQuery, [userId])
+      .then(({ rows }) => {
+        const pixScoreByCompetence = _.map(rows, (pixEarnedForOneCompetence) => pixEarnedForOneCompetence.earnedPix);
```

Comment:
Si au final on ne veut qu'une colonne, l'utilisation de la méthode `pluck` de `knex` serait probablement plus appropriée.

Reply:
`pluck` ne semble pas pouvoir être utilisé pour récupérer une colonne alias: ` column "earnedPix" does not exist`

Reply:
on a testé `.pluck(db.raw('earnedPix'))` suggéré sur cette issue https://github.com/knex/knex/issues/817, mais ça n'a pas fonctionné